### PR TITLE
RHICOMPL-217 - Metadata links hide previous/next on first/last page

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

File: app/controllers/concerns/metadata.rb [hunk 1]
```diff
@@ -17,20 +17,21 @@ def metadata(opts = {})
...
         "/#{controller_name}"
-      last_offset = (count / pagination_limit.to_f).ceil
       {
         first: "#{base_url}?limit=#{pagination_limit}&offset=1",
         last: "#{base_url}?limit=#{pagination_limit}&offset=#{last_offset}",
-        previous: previous_link(base_url),
-        next: next_link(base_url, last_offset)
-      }
+        previous: (previous_link(base_url) if pagination_offset != 1),
+        next: (if pagination_offset != last_offset
```
Comment (akofink) [inline app/controllers/concerns/metadata.rb]: We should probably use `>` here.

Comment (akofink) [reply app/controllers/concerns/metadata.rb]: Example:

```
HTTP/1.1 200 OK
{
    "data": [],
    "links": {
        "first": "/api/compliance/rules?limit=2&offset=1",
        "last": "/api/compliance/rules?limit=2&offset=514",
        "next": "/api/compliance/rules?limit=2&offset=514",
        "previous": "/api/compliance/rules?limit=2&offset=517"
    },
    "meta": {
        "limit": 2,
        "offset": 518,
        "search": null,
        "total": 1027
    }
}
```

File: app/controllers/concerns/metadata.rb [hunk 2]
```diff
@@ -17,20 +17,21 @@ def metadata(opts = {})
...
       base_url = "#{ENV['PATH_PREFIX']}/#{ENV['APP_NAME']}"\
         "/#{controller_name}"
-      last_offset = (count / pagination_limit.to_f).ceil
       {
         first: "#{base_url}?limit=#{pagination_limit}&offset=1",
         last: "#{base_url}?limit=#{pagination_limit}&offset=#{last_offset}",
-        previous: previous_link(base_url),
-        next: next_link(base_url, last_offset)
-      }
+        previous: (previous_link(base_url) if pagination_offset > 1),
```
Comment (akofink) [inline app/controllers/concerns/metadata.rb]: This is nefarious of me, but the previous link should not ever link to an invalid/empty page.

```
$ http --verify no https://ci.foo.redhat.com:1337/api/compliance/rules limit==2 offset==518
{
    "data": [],
    "links": {
        "first": "/api/compliance/rules?limit=2&offset=1",
        "last": "/api/compliance/rules?limit=2&offset=514",
        "previous": "/api/compliance/rules?limit=2&offset=517"
    },
    "meta": {
        "limit": 2,
        "offset": 518,
        "search": null,
        "total": 1027
    }
}
```

Comment (dLobatog) [reply app/controllers/concerns/metadata.rb]: :smile:

**Pull request merged**